File: backend/src/main/java/com/cduestc/openatom/project/entity/Project.java
package com.cduestc.openatom.project.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("projects")
public class Project {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    
    private String name;
    private String description;
    private String coverImage;
    private String repositoryUrl;
    private String demoUrl;
    private Long leaderId;
    private String status; // recruiting, developing, completed
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    @TableLogic
    private Boolean deleted;
}

File: frontend/src/types/project.ts
export interface Project {
  id: number
  name: string
  description: string
  coverImage: string
  repositoryUrl: string
  demoUrl: string
  leaderId: number
  leaderName?: string
  status: string // recruiting, developing, completed
  createdAt: string
  updatedAt: string
}

export interface ProjectQuery {
  page: number
  size: number
  status?: string
  keyword?: string
}

export interface ProjectMember {
  id: number
  projectId: number
  userId: number
  userName?: string
  role: string
  joinedAt: string
}
